### PR TITLE
WinUI `TeachingTip`

### LINKED ISSUE #3911: WinUI `TeachingTip`
## What would you like to be added:

WinUI `TeachingTip` control.

## Why is this needed:

WinUI parity.

## For which Platform:

- [x] iOS
- [x] Android
- [x] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [x] macOS
- [x] Skia
  - [x] WPF
  - [x] GTK (Linux)
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes #3911

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

Not supported.

## What is the new behavior?

image

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/TeachingTip/TeachingTip.mux.cs [hunk 1]
```diff
@@ -0,0 +1,2673 @@
...
+		m_previewKeyDownForF6Revoker.Disposable = null;
+		m_effectiveViewportChangedRevoker.Disposable = null;
+		m_contentSizeChangedRevoker.Disposable = null;
+		m_closeButtonClickedRevoker.Disposable = null;
+		m_alternateCloseButtonClickedRevoker.Disposable = null;
+		m_actionButtonClickedRevoker.Disposable = null;
+
+		m_container = (Border)GetTemplateChild(s_containerName);
+		m_rootElement = m_container.Child;
+		m_rootElement.FreezeTemplatedParent();
```
Comment (Youssef1313) [inline src/Uno.UI/Microsoft/UI/Xaml/Controls/TeachingTip/TeachingTip.mux.cs:73]: Let's add an "Uno specific" comment with link issue so we can remove this workaround when the issue is fixed in future.

File: src/Uno.UI/UI/Xaml/FrameworkElement.Android.cs [hunk 2]
```diff
@@ -65,6 +65,7 @@ private void PerformOnLoaded(bool isFromResources = false)
 				{
 					if (resource is FrameworkElement resourceAsFrameworkElement)
 					{
+						resourceAsFrameworkElement.XamlRoot = XamlRoot;
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/FrameworkElement.Android.cs:68]: Is it set to `null` on `Unloaded`?

File: src/Uno.UI/UI/Xaml/UIElement.crossruntime.cs [hunk 3]
```diff
@@ -131,6 +131,7 @@ private void OnElementLoaded()
 				{
 					if (resource is FrameworkElement resourceAsUIElement)
 					{
+						resourceAsUIElement.XamlRoot = XamlRoot;
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/UIElement.crossruntime.cs:134]: Same here, is it set back to `null` on Unloaded?

**Review by Youssef1313 — APPROVED** [review]

File: src/Uno.UI/UI/Xaml/FrameworkElement.Android.cs (hunk 2 above)
Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/FrameworkElement.Android.cs:68]: Will do 👍

File: src/Uno.UI/UI/Xaml/UIElement.crossruntime.cs (hunk 3 above)
Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/UIElement.crossruntime.cs:134]: Actually realized that should not happen - once an element is associated with a `XamlRoot`, it must stay there - 
image

**Pull request merged**